index.bs

```diff
@@ -1877,6 +1877,37 @@ If the bytes in the buffer have a natural intepretation
 as one of the other TypedArray types, provide that instead.
 For example, if the bytes represent Float32 values, use a {{Float32Array}}.
 
+<h3 id="allow-microtask-switches">A Promise represents a value, not a callback</h3>
+
+Avoid requiring that something needs to happen synchronously within
+a promise resolution or rejection callback.
+
+Authors take for granted that an `async` function can be wrapped
+in another that appends synchronous code at the end e.g. to inspect state
+using try/catch. But the `await` queues a microtask:
+<div class="example">
+```js
+async function fooWrapper() {
+  const foo = await platform.foo();
+  console.log("success");
+  return foo;
+}
+
+(async () => {
+  const foo = await fooWrapper();
+  foo.bar(); // on the same task, but not the same microtask that logged "success"
+})();
+```
+</div>
+
+To not interfere with this pattern, the lowest recommended restriction is "same task".
+
+<div class="note">
+One case where this came up was the [captureController.setFocusBehavior()](https://w3c.github.io/mediacapture-screen-share/#dom-capturecontroller-setfocusbehavior)
+method, where timing concerns were solved by adding an timeout on top of
+the same task requirement.
```

Comment:
I don't see a same task requirement in https://w3c.github.io/mediacapture-screen-share/#dom-capturecontroller-setfocusbehavior - please clarify.
setFocusBehavior() as currently specified doesn't return a promise.

Reply:
setFocusBehavior() throws outside of the getDisplayMedia success task. See [wpt](https://searchfox.org/mozilla-central/rev/1d1ca77dd7ca10953c6f5f1b77b98e4eb528274e/testing/web-platform/tests/screen-capture/getdisplaymedia-capture-controller.https.window.js#150-151,153,156,158-160) and [fyi](https://wpt.fyi/results/screen-capture/getdisplaymedia-capture-controller.https.window.html?label=experimental&label=master&aligned).

The spec could use an example.

Reply:
Aha. In order to find that, one has to find the link from "finalize focus decision algorithm" to "getDisplayMedia".
Agreed that the spec could be easier to comprehend if an example was added - or just text that states the purpose of the mechanism.

So setFocusBehavior() is an example of a function that must be called in the same task as getDisplayMedia's resolution, but the description never uses the words "same task".

Reply:
setFocusBehavior() queues tasks to implement this (details fixed in https://github.com/w3c/mediacapture-screen-share/issues/303).

Other specs [use](https://w3c.github.io/mediacapture-extensions/#dfn-expose-video-frame-counters-steps) _"Let task be a the current [task](https://html.spec.whatwg.org/multipage/webappapis.html#concept-task). If [[[LastTask]]](https://w3c.github.io/mediacapture-extensions/#dfn-lasttask-0) is equal to task..."_. They accomplish similar things.